### PR TITLE
Vega 1014 cond headers

### PR DESCRIPTION
**Pull request by JosephMcDonagh**
Added conditional header field to mte config

### CODE REVIEW

File: internal/provider/config_resource.go [hunk 1]
```diff
@@ -421,5 +475,21 @@ func transformToResourceModel(d *client.MTEConfigDto) MTEConfigModel {
 		}
 	}
 
+	if len(d.ConditionalHeaders) != 0 {
+
```
Comment (Ashutosh2302) [inline internal/provider/config_resource.go]: This logic appears to be some duplication of logic on line 410, can we perhaps extract this to a function?

Comment (JosephMcDonagh) [reply internal/provider/config_resource.go]: have refactored to match the architecture of the other fields in the resource and api models, see, what you think of that

File: internal/provider/config_resource.go [hunk 2]
```diff
@@ -214,6 +214,28 @@ type ConditionalHeaderModel struct {
...
+		NewHeader:      c.NewHeader.ValueString(),
+		MatchValue:     c.MatchValue.ValueString(),
+		NoMatchValue:   c.NoMatchValue.ValueString(),
+	}
+	return condHeadPostBody
+}
+
+func transformCondHeaderToResourceModel(c *client.ConditionalHeaderDto) ConditionalHeaderModel {
+	var condHeaderResourceModel = ConditionalHeaderModel{
+		MatchingHeader: types.StringValue(c.MatchingHeader),
```
Comment (Ashutosh2302) [inline internal/provider/config_resource.go:230]: ConditionalHeaderDto and ConditionalHeaderModel have the same fields, is it possible just to use one of these?

Comment (JosephMcDonagh) [reply internal/provider/config_resource.go:230]: I believe they need to be two separate models due to the way the go code understands them internally, the annotations on the end of each of the types, one interchanges from json to a go struct and one is the same but between go struct and a tf resource.

**Review by Ashutosh2302 — APPROVED** [review]

**Pull request merged**